runtime-modules/content/src/types.rs

```diff
@@ -576,13 +554,18 @@ impl<ChannelId: Clone, SeriesId: Clone, VideoPostId: Clone, OwnedNFT: Clone>
 }
 
 /// Operations with local pallet account.
-pub trait ModuleAccount<T: balances::Trait> {
+pub trait ModuleAccount<T: Trait> {
     /// The module id, used for deriving its sovereign account ID.
     type ModuleId: Get<ModuleId>;
 
     /// The account ID of the module account.
     fn module_account_id() -> T::AccountId {
-        Self::ModuleId::get().into_sub_account(Vec::<u8>::new())
+        Self::ModuleId::get().into_sub_account(T::ChannelId::zero())
+    }
+
+    /// The account ID for the specified channel_id
+    fn account_for_channel_id(channel_id: T::ChannelId) -> T::AccountId {
+        Self::ModuleId::get().into_sub_account(channel_id)
```

Comment:
What about if `channel_id`== 0? wouldn't this give same account as fore the entire module?
Wouldn't it be a bit more safe to have something like `"channel"||channel_id` to make it impossible for there to be conflict?

Reply:
see above comment